pkgs/development/ocaml-modules/pcre/default.nix

```diff
@@ -2,15 +2,15 @@
 
 buildDunePackage rec {
   pname = "pcre";
-  version = "7.4.6";
+  version = "7.5.0";
 
   useDune2 = true;
 
-  minimalOCamlVersion = "4.08";
+  minimalOCamlVersion = "4.12";
```

Comment:
This change removes quite a few packages (for OCaml < 4.12). How bad is this?

Reply:
in nixpkgs it seems fine, for external consumers that might be different (though that's always different).

The only alternative is packaging the old one as ocaml_pcre_746 or something. But that will still break.

I'm not sure why the ocaml subsystem in nixpkgs "supports" such a wide version range anyways though. It's quite different from how other languages in nixpkgs do things.

I might be missing some details but having a policy for generally only supporting the last two minor versions similar to python seems reasonable to me.

Anyways, should I be bumping the version constraint for the library consumers as well in this case?

Reply:
Thanks for your reply.

> Anyways, should I be bumping the version constraint for the library consumers as well in this case?

I think it’s fine.